src/actions/index.js

```diff
@@ -43,69 +43,118 @@ export function setProjectList(projectList) {
   return { type: SET_PROJECT_LIST, projectList }
 }
 
-export function storeUser(user) {
-  return () => {
-    store.save('@zooniverse:user', {
+export function syncUserStore() {
+  return (dispatch, getState) => {
+    const user = getState().user
+    return store.save('@zooniverse:user', {
       user
     })
   }
 }
 
 export function setUserFromStore() {
   return dispatch => {
-    dispatch(setIsFetching(true))
-    store.get('@zooniverse:user')
-      .then(json => {
+    return new Promise ((resolve, reject) => {
+      store.get('@zooniverse:user').then(json => {
         dispatch(setUser(json.user))
-        dispatch(setIsFetching(false))
+        return resolve()
+      }).catch(() => {
+        return reject()
       })
-      .catch(() => { //nothing here, send user to login screen
-        Actions.SignIn()
-        dispatch(setIsFetching(false))
-      });
+    })
   }
 }
 
 export function continueAsGuest() {
   return dispatch => {
-    var user = { isGuestUser: true }
     dispatch(setState('user.isGuestUser', true))
-    dispatch(storeUser(user))
+    dispatch(syncUserStore())
     Actions.ZooniverseApp({type: ActionConst.RESET})
   }
 }
 
+export function checkIsConnected() {
+  return () => {
+    return new Promise((resolve, reject) => {
+      NetInfo.isConnected.fetch().then(isConnected => {
+        if (!isConnected) {
+          return reject('Sorry, but you must be connected to the internet to use Zooniverse')
+        }
+        return resolve()
+      })
+    })
+  }
+}
+
 export function signIn(login, password) {
   return dispatch => {
     dispatch(setIsFetching(true))
     dispatch(setError(''))
-    NetInfo.isConnected.fetch().then(isConnected => {
-      if (isConnected) {
-        auth.signIn({login: login, password: password})
-          .then((user) => {
-            user.get('avatar')
-              .then((avatar) => {
-                user.avatar = head(avatar)
-              })
-              .catch(() => {
-                user.avatar = {}
-              })
-              .then(() => {
-                user.isGuestUser = false
-                dispatch(setUser(user))
-                dispatch(storeUser(user))
-                dispatch(setIsFetching(false))
-                Actions.ZooniverseApp({type: ActionConst.RESET})
-              })
-          })
-          .catch((error) => {
-            dispatch(setError(error.message))
-            dispatch(setIsFetching(false))
-          })
-      } else {
-        dispatch(setError('Sorry, but you must be connected to the internet to use Zooniverse'))
+    dispatch(checkIsConnected()).then(() => {
+      auth.signIn({login: login, password: password}).then((user) => {
+        user.isGuestUser = false
+        dispatch(setUser(user))
+
+        return Promise.all([
+          dispatch(loadUserAvatar()) //will have more added
```

Comment:
What will be added here? It seems a bit awkward to use `Promise.all` with an array of length one.